src/GraphQL.Client.Serializer.SystemTextJson/ImmutableConverter.cs

```diff
@@ -19,7 +19,7 @@ public override bool CanConvert(Type typeToConvert)
                 return false;
 
             var nullableUnderlyingType = Nullable.GetUnderlyingType(typeToConvert);
-            if (nullableUnderlyingType != null && nullableUnderlyingType.IsPrimitive)
+            if (nullableUnderlyingType != null && nullableUnderlyingType.IsValueType)
```

Comment:
😄

Reply:
What exactly do you mean by this review? 😄

Reply:
The devil is in the little things.